reco_utils/dataset/amazon_reviews.py

```diff
@@ -233,6 +242,87 @@ def _data_generating(input_file, train_file, valid_file, test_file, min_sequence
             cate_list.append(category)
             dt_list.append(date_time)
 
+def _data_generating_no_history_expanding(input_file, train_file, valid_file, test_file, min_sequence=1):
+    '''produce train, valid and test file from processed_output file
+    Each user's behavior sequence will only produce one line in trian file.
+    Like, user's behavior sequence: 12345, and this function will write into train file: 12345
+    '''
+    f_input = open(input_file, "r")
+    f_train = open(train_file, "w")
+    f_valid = open(valid_file, "w")
+    f_test = open(test_file, "w")
+    print("data generating...")
+
+    last_user_id = None
+    last_movie_id = None
+    last_category = None
+    last_datetime = None
+    last_tfile = None
+    for line in f_input:
+        line_split = line.strip().split("\t")
+        tfile = line_split[0]
+        label = int(line_split[1])
+        user_id = line_split[2]
+        movie_id = line_split[3]
+        date_time = line_split[4]
+        category = line_split[5]
+
+        if last_tfile == "train":
+            fo = f_train
+        elif last_tfile == "valid":
+            fo = f_valid
+        elif last_tfile == "test":
+            fo = f_test
+        if user_id != last_user_id or tfile == "valid" or tfile == "test":
+            if last_user_id != None:
```

Comment:
To check if none or not, we use `is not None`
> PEP8: Comparisons to singletons like None should always be done with is or is not, never the equality operators.

Reply:
good catch!